wandb/sdk/wandb_run.py

```diff
@@ -1010,134 +1005,118 @@ def job_type(self) -> str:
-            root = os.path.abspath(root)
-            for file_path in filenames.filtered_dir(root, include_fn, exclude_fn):
-                files_added = True
-                save_name = os.path.relpath(file_path, root)
-                art.add_file(file_path, name=save_name)
-        # Add any manually staged files such as ipynb notebooks
-        for dirpath, _, files in os.walk(self._settings._tmp_code_dir):
-            for fname in files:
-                file_path = os.path.join(dirpath, fname)
-                save_name = os.path.relpath(file_path, self._settings._tmp_code_dir)
-                files_added = True
-                art.add_file(file_path, name=save_name)
-        if not files_added:
-            wandb.termwarn(
-                "No relevant files were detected in the specified directory. No code will be logged to your run."
-            )
-            return None
-
-        return self._log_artifact(art)
+        deprecate.deprecate(
+            field_name=deprecate.Deprecated.run__get_project_url,
+            warning_message=(
+                "The get_project_url method is deprecated and will be removed in a"
+                " future release. Please use `run.project_url` instead."
+            ),
+        )
+        return self.project_url
 
-    def get_project_url(self) -> str | None:
+    @property
+    @_run_decorator._attach
+    def project_url(self) -> str | None:
         """Return the url for the W&B project associated with the run, if there is one.
 
         Offline runs will not have a project url.
         """
         if self._settings._offline:
             wandb.termwarn("URL not available in offline run")
             return None
+        if self._settings._noop:
+            return None
         return self._settings.project_url
 
     def get_sweep_url(self) -> str | None:
-        """Return the url for the sweep associated with the run, if there is one."""
+        """Return the url for the sweep associated with the run, if there is one.
+
+        Offline runs will not have a sweep url.
+
+        Note: this method is deprecated and will be removed in a future release.
+        Please use `run.sweep_url` instead.
+        """
+        deprecate.deprecate(
+            field_name=deprecate.Deprecated.run__get_sweep_url,
+            warning_message=(
+                "The get_sweep_url method is deprecated and will be removed in a"
+                " future release. Please use `run.sweep_url` instead."
+            ),
+        )
+        return self.sweep_url
+
+    @property
+    @_run_decorator._attach
+    def sweep_url(self) -> str | None:
+        """Return the url for the sweep associated with the run, if there is one.
+
+        Offline runs will not have a sweep url.
+        """
         if self._settings._offline:
             wandb.termwarn("URL not available in offline run")
             return None
+        if self._settings._noop:
+            return None
```

Comment:
Here too, what is the purpose of this?

Reply:
to be backward compatible this was the behavior before, so i just kept it (same for other instances you asked)

Reply:
It doesn't look like this was the behavior before, am I missing something?

Reply:
see this test: https://github.com/wandb/wandb/pull/8925/files#diff-624e792ad1626d283aef9fc0758454a8f2e606618f6af7de8955df11361bae0aR77

Reply:
How was it implemented before? The old implementation of `get_sweep_url` doesn't look like it checks `_noop`.

Reply:
i can dig it up, but you can see that the tests that i didn't change assert that it should be none when in disabled mode, so if i remove these lines in all instances it will break

Reply:
ok i found it, it is done when you set the run to disabled these functions are by default assigned None as the return value see here: https://github.com/wandb/wandb/blob/main/wandb/sdk/wandb_init.py#L710-L730

Reply:
Then shouldn't we update that code instead for consistency?

Reply:
sure i can do that instead